### PR TITLE
update enum path, log and unit test

### PR DESCRIPTION
**Pull request by lxning**

### CODE REVIEW

**Comment by shz117** [comment]
it's more like "add logging and metrics for etcd integration"..
can you rename the PR or add some context in the description the help ppl review?

File: subscriber/common/job/controller.go [hunk 1]
```diff
@@ -143,27 +143,29 @@ func NewController(params Params) *Controller {
 		params.ServiceConfig.Logger.Info("aresDB Controller is enabled")
 
 		if params.ServiceConfig.HeartbeatConfig.Enabled {
+			params.ServiceConfig.Logger.Info("heartbeat config",
+				zap.Any("heartbeat", *params.ServiceConfig.HeartbeatConfig))
 			controller.etcdServices, err = connectEtcdServices(params)
 			if err != nil {
 				panic(utils.StackError(err, "Failed to createEtcdServices"))
```
Comment (jshencode) [inline subscriber/common/job/controller.go]: better use logger.fatal so we can catch the log in elk

File: subscriber/common/job/controller.go [hunk 2]
```diff
@@ -143,27 +143,29 @@ func NewController(params Params) *Controller {
...
-		err = controller.zkClient.Start()
-		if err != nil {
-			panic(utils.StackError(err, "Failed to start zkClient"))
-		}
-		params.ServiceConfig.Logger.Info("zkClient was started")
+		} else {
+			controller.zkClient = createZKClient(params)
+			err = controller.zkClient.Start()
+			if err != nil {
+				panic(utils.StackError(err, "Failed to start zkClient"))
```
Comment (jshencode) [inline subscriber/common/job/controller.go]: same here

File: subscriber/common/job/controller.go [hunk 3]
```diff
@@ -143,27 +143,29 @@ func NewController(params Params) *Controller {
...
+			}
+			params.ServiceConfig.Logger.Info("zkClient was started")
 
-		err = controller.RegisterOnZK()
-		if err != nil {
-			panic(utils.StackError(err, "Failed to register subscriber"))
+			err = controller.RegisterOnZK()
+			if err != nil {
+				panic(utils.StackError(err, "Failed to register subscriber"))
```
Comment (jshencode) [inline subscriber/common/job/controller.go]: same here

File: subscriber/common/job/controller.go (hunk 1 above)
Comment (lxning) [reply subscriber/common/job/controller.go]: here should stop the service to alert deployment failed via panic.

File: subscriber/common/job/controller.go (hunk 3 above)
Comment (lxning) [reply subscriber/common/job/controller.go]: same here.

File: subscriber/common/job/controller.go [hunk 4]
```diff
@@ -18,13 +18,13 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"github.com/m3db/m3/src/cluster/placement"
```
[the remark this answers is not in the document]
Comment (lxning) [reply subscriber/common/job/controller.go]: yes, from auto fmt.

File: subscriber/common/job/controller.go (hunk 1 above)
Comment (lxning) [reply subscriber/common/job/controller.go]: replace with logger.panic

File: subscriber/common/job/controller.go (hunk 2 above)
Comment (lxning) [reply subscriber/common/job/controller.go]: same

File: subscriber/common/job/controller.go (hunk 3 above)
Comment (lxning) [reply subscriber/common/job/controller.go]: same

**Pull request merged**